fix private tag conversion for elements == 0xff

Johannes-sg opened the pull request:
Fixes #1059  .

#### Checklist
- [x] The pull request branch is in sync with latest commit on the *fo-dicom/development* branch
- [ ] I have updated API documentation
- [x] I have included unit tests
- [x] I have updated the change log
- [x] I am listed in the CONTRIBUTORS file

#### Changes proposed in this pull request:
- Fixing private tag conversion for case that element equals 0xff

shazha commented:
One of the best practices for writing unit tests is avoid multiple asserts in a single test. It's better to use `Theory` and `InlineData` in this case😄. Microsoft provides more [examples](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-multiple-asserts) and guidance on how to write unit tests.

Johannes-sg commented:
> One of the best practices for writing unit tests is avoid multiple asserts in a single test. It's better to use `Theory` and `InlineData` in this case😄. Microsoft provides more [examples](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#avoid-multiple-asserts) and guidance on how to write unit tests.

OK, I changed it now. I saw the Theory and InlineData yesterday, but did not find it suitable in my case. The test logic is now changed a little bit in order to avoid multiple asserts statements, but still tests what it needs to :)
